src/main/kotlin/org/malachite/estella/services/JobSeekerService.kt

```diff
@@ -34,6 +36,22 @@ class JobSeekerService(
             throw UserAlreadyExistsException()
         }
 
+    fun deleteJobSeeker(id: Int, jwt: String?) =
+        if(!checkRights(id, jwt).contains(Permission.DELETE)) throw UnauthenticatedException()
+        else deleteJobSeeker(id)
+
+    private fun deleteJobSeeker(id: Int) = jobSeekerRepository.deleteById(id)
+
+    private fun checkRights(id: Int, jwt: String?): Set<Permission> {
+        if(securityService.isCorrectApiKey(jwt)) return Permission.allPermissions()
+        securityService.getJobSeekerFromJWT(jwt)
+            ?.let {
+                if(id == it.id) return Permission.allPermissions()
+                else throw UnauthenticatedException()
+            }
+            ?: throw UnauthenticatedException()
```

Comment:
Else i dwa razy rzucany wyjątek nie dałoby się tego obrobić tak że tylko raz byłby rzucany wyjątek?

Reply:
teoretycznie mogłoby być else null ale nie wiem, czy to nie jest cheaterskie